lib/find-file.js

```diff
@@ -53,18 +53,14 @@ function findFile(opt) {
       // find path to the root, stopping at cwd, package.json or bower.json
       var pathToRoot = [];
       var isWorking;
-      var depthCount = 0;
       do {
         pathToRoot.push(absoluteStart);
-        if (!testNotPackage(absoluteStart)) {
-          depthCount += 1;
-        }
-        isWorking = testWithinLimit(absoluteStart) && depthCount < options.moduleDepth;
+        isWorking = testWithinLimit(absoluteStart) && testNotPackage(absoluteStart);
         absoluteStart = path.resolve(absoluteStart, '..');
       } while (isWorking);
 
       // start a queue with the path to the root
-      var queue = pathToRoot.concat();
+      var queue = pathToRoot.concat(!!options.root && pathToRoot.indexOf(options.root) === -1 ? limit : []);
```

Comment:
How do you like this option:
```
// start a queue with the path to the root
var queue = pathToRoot.concat(pathToRoot.indexOf(limit) === -1 ? limit : []);
```
Addition option like `options.includeRoot ` is excess. If we have `options.root` with real value than `options.includeRoot ` must be `true` by default. In another case, if `options.root: null` than `limit = process.cwd()` and, again, root is must be included/allowed by default.

In the current situation, it seems to me that it is sufficient to check whether there is a `limit` in the `pathToRoot` .